src/cache/requests/list/list_fetch.rs

```diff
@@ -0,0 +1,225 @@
+/// # fn main() -> anyhow::Result<()> {
+/// # use momento_test_util::create_doctest_cache_client;
+/// # tokio_test::block_on(async {
+/// use std::convert::TryInto;
+/// use momento::cache::{ListFetch, ListFetchRequest};
+/// use momento::MomentoErrorCode;
+/// # let (cache_client, cache_name) = create_doctest_cache_client();
+/// let list_name = "list-name";
+/// # cache_client.list_concatenate_front(&cache_name, list_name, vec!["value1", "value2"]).await;
+///
+/// let fetch_request = ListFetchRequest::new(cache_name, list_name).start_index(1).end_index(3);
+/// let fetched_values: Vec<String> = cache_client.send_request(fetch_request).await?.try_into().expect("Expected a list fetch!");
+/// # Ok(())
+/// # })
+/// # }
+/// ```
+pub struct ListFetchRequest<L: IntoBytes> {
+    cache_name: String,
+    list_name: L,
+    start_index: Option<i32>,
+    end_index: Option<i32>,
+}
+
+impl<L: IntoBytes> ListFetchRequest<L> {
+    pub fn new(cache_name: impl Into<String>, list_name: L) -> Self {
+        Self {
+            cache_name: cache_name.into(),
+            list_name,
+            start_index: None,
+            end_index: None,
+        }
+    }
+
+    /// Set the starting inclusive element of the list to fetch.
+    pub fn start_index(mut self, start_index: i32) -> Self {
+        self.start_index = Some(start_index);
+        self
+    }
+
+    /// Set the ending exclusive element of the list to fetch.
+    pub fn end_index(mut self, end_index: i32) -> Self {
+        self.end_index = Some(end_index);
+        self
+    }
+}
+
+impl<L: IntoBytes> MomentoRequest for ListFetchRequest<L> {
+    type Response = ListFetch;
+
+    async fn send(self, cache_client: &CacheClient) -> MomentoResult<ListFetch> {
+        let start_index = match self.start_index {
+            Some(start) => Some(StartIndex::InclusiveStart(start)),
+            None => Some(StartIndex::UnboundedStart(Unbounded {})),
+        };
+        let end_index = match self.end_index {
+            Some(end) => Some(EndIndex::ExclusiveEnd(end)),
+            None => Some(EndIndex::UnboundedEnd(Unbounded {})),
+        };
+        let request = prep_request_with_timeout(
+            &self.cache_name,
+            cache_client.configuration.deadline_millis(),
+            momento_protos::cache_client::ListFetchRequest {
+                list_name: self.list_name.into_bytes(),
+                start_index,
+                end_index,
+            },
+        )?;
+
+        let response = cache_client
+            .data_client
+            .clone()
+            .list_fetch(request)
+            .await?
+            .into_inner();
+
+        match response.list {
+            Some(list_fetch_response::List::Missing(_)) => Ok(ListFetch::Miss),
+            Some(list_fetch_response::List::Found(found)) => Ok(ListFetch::Hit {
+                values: ListFetchValue {
+                    raw_item: found.values,
+                },
+            }),
+            _ => unreachable!(),
```

Comment:
In the sorted set implementation we had a case for None, which we interpreted as a `Miss`

Reply:
I could return an unknown error instead like so? (and update accordingly for other instances as well)

```
MomentoError {
    message: "Unknown error has occurred, unable to parse list_fetch_response".into(),
    error_code: MomentoErrorCode::UnknownError,
    inner_error: Some(response),
    details: None,
}
```

Reply:
Yes that looks good. That follows the JS pattern [here](https://github.com/momentohq/client-sdk-javascript/blob/b0ed1b3ac745ee1eeca1cc8cbc672b1a151268cc/packages/client-sdk-nodejs/src/internal/cache-data-client.ts#L2505-L2515) and [here](https://github.com/momentohq/client-sdk-javascript/blob/main/packages/client-sdk-nodejs/src/errors/cache-service-error-mapper.ts#L99). We can extract this to a function later.

Reply:
Made a ticket to do that throughout the sdk: https://github.com/momentohq/client-sdk-rust/issues/239